src/controller/CHIPDeviceController.cpp

```diff
@@ -1040,62 +1041,58 @@ void DeviceCommissioner::OnSessionEstablishmentTimeoutCallback(System::Layer * a
 {
     reinterpret_cast<DeviceCommissioner *>(aAppState)->OnSessionEstablishmentTimeout();
 }
-
+#if CHIP_DEVICE_CONFIG_ENABLE_MDNS
 CHIP_ERROR DeviceCommissioner::DiscoverAllCommissioning()
 {
-#if CHIP_DEVICE_CONFIG_ENABLE_MDNS
+    for (int i = 0; i < kMaxCommissionableNodes; ++i)
+    {
+        mCommissionableNodes[i].Reset();
+    }
```

Comment:
This is racy if called multiple times before the end. There should be a busy error if called again when resolver already running.

Reply:
The question is...what is "running". mDNS advertisers are free to send new information at any point, and that would add information to the list. The proper way to do this is to obey the TTLs in the messages and have a timer periodically check as they get stale. That's a larger project than what I'm attempting to do here, so another reviewer asked for the reset to be added. Locking this while we expect devices to be sending back info is a guessing game at times. Can we compromise on these and just make an explicit clear function until the TTL stuff is in?

Reply:
Running means that DiscoveryAllCommissioning() was called. It's not going to run forever. While it's running, multiple threads can be mutating `mCommissionableNodes`. That's what I am asking to guard against.

It's a separate concern the fact that the lifecycle of when a client might want to discover all commissioning-enabled nodes. The issue right now is that you might be able to call it from multiple contexts and have access races, especially on mobiles or Linux where the CPUs are multicore and the CHIP main task that processes incoming minmdns packets is different than the calling task of application logic trying to discover nodes.